### ISSUE TITLE
Use lazy loading for displayMode builder variables 

### ISSUE THREAD

**Issue by renickbuettner** [rainlab/pages-plugin#499]
- OctoberCMS Build: 2.2.15
- RainLab Pages Plugin Version: 1.14.12
- PHP Version: 8.0.1

### Description:
We're using static page variables to build modular, re-usable layouts/ pages. Since there is the new displayMode builder, the ui for sorting really much fields is way better. We're using repeater groups to split elements into blocks and are able to arrange them freely.

Really much fields within the repeater, there are two problems. The first is, that it transfers lots of input variables. We already increase the php input vars limit to 2500 or above. Next to that, the ui handles all fields at once. That makes the ui really slow on content-heavy pages, especially on slower computers. For example do we have clients with Windows computers, which have troubles using the ui. May it would be possible to just keep the current repeater group within the memory. 

_Not sure if this is something relating to Static Pages rather than October forms. We also have implemented plugins which store content heavy sites within the database - and there's the same problem when having many fields._

### tl;dr
- really much input variables because the complete form is submitted everytime
- really slow ui because all fields stay rendered in memory

### Steps To Reproduce:
1. Add a page variable (repeater) with groups. We often use it as "blocks" or "content_sections".
2. Have group elements like section to define content sections.
3. The section has a repeater field "content" which contains content elements. This elements can be paragraph, gallery, grid, call-to-action, video, integration, database widgets...
4. Add lots of content like 8 sections with at least 8 to 12 elements.

**Comment by daftspunk** [comment]
Hi @renickbuettner 

We are working on a solution to this called [Tailor](https://portal.octobercms.com/c/22-tailor). At the moment the Pages plugin tries to lump all the data types into one object and then that itself is wrapped in a SPA. This inefficiency is expressed as a slow user interface at scale. With Tailor, a lot of these structures can be broken down into smaller objects and assembled in a final page, either in the theme (using components) or by a parent object (using relationships). We are working hard to get this release ready for October v2.3 so stay tuned.

Regarding the lazy loading idea for the builder mode, I've added this to our internal tracker. Thanks for this suggestion.
